scripts/src/main/resources/scripts/functions

```diff
@@ -1810,53 +1816,138 @@ function doGetSoftwareVersion() {
   fi
 }
 
+# $1: software
+function doGetSoftwareEdition() {
+  local software="${1}"
+  local editionVar
+  editionVar="$(echo "${software/-/_}" | awk '{print toupper($0)}')_EDITION"
+  local edition
+  edition="${!editionVar}"
+  if [ -z "${edition}" ]
+  then
+    edition="${software}"
+  fi
+  echo "${edition}"
+}
+
+# $1: software
+# $2: optional prefix
+function doGetLatestSoftwareVersion() {
+  local software="${1}"
+  local prefix="${2}"
+  local edition
+  local version
+  doUpdateUrls
+  edition=$(doGetSoftwareEdition "${software}")
+  if [ -z "${prefix}" ]
+  then
+    version="$(find "${DEVON_IDE_HOME}"/urls/"${software}"/"${edition}" -mindepth 1 -maxdepth 1 -print | awk -F'/' '{print $NF}' | sort -rV | head -1)"
+  else
+    prefix="${prefix:0:${#prefix}-1}"
+    prefix="${prefix/./[.]}"
+    version="$(find "${DEVON_IDE_HOME}"/urls/"${software}"/"${edition}" -mindepth 1 -maxdepth 1 -print | awk -F'/' '{print $NF}' | grep "^${prefix}" | sort -rV | head -1)"
+  fi
+  echo "${version}"
+} 
+
 # $1: software
 # $2: selected version
 # $3: commandlet name
 function doSetSoftwareVersion() {
   local software="${1}"
   local version="${2}"
   local commandlet="${3}"
-  if [ -n "${version}" ]
+  local edition
+  if [ -d "${DEVON_IDE_HOME}/urls" ]
   then
-    if [ "${version}" = "latest" ]
+    if [ -n "${version}" ]
     then
-      doUpdateMirrors
-      version="$(head -n 1 "${DEVON_IDE_HOME}/mirrors/${software}/available-versions")"
+      if [ "${version}" = "latest" ]
+      then
+        doUpdateUrls
+        version=$(doGetLatestSoftwareVersion "${software}")
+      fi
+      local software_version_variable="${software}"
+      # temporary workaround for #1098 should be fixed and removed with #1085
+      if [ "${software_version_variable}" = "nodejs" ]
+      then
+        software_version_variable="node"
+      fi
+      software_version_variable="$(echo "${software_version_variable}" | awk '{print toupper($0)}')_VERSION"
+      local devon_properties="${DEVON_IDE_HOME}/settings/devon.properties"
+      grep -v "${software_version_variable}" "${devon_properties}" > "${devon_properties}.tmp"
+      mv "${devon_properties}.tmp" "${devon_properties}"
+      echo "${software_version_variable}=${version}" >> "${devon_properties}"
+      doEchoInteraction "${software_version_variable}=${version} has been set in ${devon_properties}\nTo install that version call the following command:\ndevon ${commandlet} setup"
+    else
+      doError "You have to specify the version you want to set."
```

Comment:
This will cause compatibility issues or you have to adapt all commandlets.
So far the empty string for version is treated the same as `latest` to make it work if $«TOOL»_VERSION is unset.
I would strongly recommend to keep this behavior and just join the ifs so if version is empty or latest, you simply first set `version` to the most recent one.